File: CS/Controllers/HtmlEditorController.cs
using System;
using System.Collections.Generic;
using System.Web.UI.WebControls;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using DevExpress.Web.ASPxHtmlEditor;

namespace CS.Controllers {
    public class HtmlEditorController : Controller {
        const string SampleDocumentPath = "~/Content/HtmlEditor/SampleImportDocument.rtf";

        public ActionResult ImportExport() {
            ViewData["SampleDocumentPath"] = SampleDocumentPath;
            string htmlContentPath = Server.MapPath("~/Content/HtmlEditor/ImportExport.htm");
            string html = System.IO.File.ReadAllText(htmlContentPath);
            return View("ImportExport", new HtmlEditorModel(html));
        }

        public ActionResult ImportExportPartial() {
            return PartialView("ImportExportPartial");
        }

        public ActionResult ExportTo(HtmlEditorExportFormat format) {
            return HtmlEditorExtension.Export(
                HtmlEditorDemosHelper.SetHtmlEditorExportSettings(new HtmlEditorSettings()),
                format
            );
        }

        /**/
        [HttpPost]
        public ActionResult ExportContentTo() {
            HtmlEditorExportFormat format = EditorExtension.GetValue<HtmlEditorExportFormat>("cmbFormat");
            
            System.IO.MemoryStream stream = new System.IO.MemoryStream();
            HtmlEditorExtension.Export(HtmlEditorDemosHelper.SetHtmlEditorExportSettings(new HtmlEditorSettings()), stream, format);

            string fileFormat = format.ToString().ToLower().TrimStart('.');

            FileStreamResult result = new FileStreamResult(stream, "application/" + fileFormat);
            result.FileDownloadName = string.Format("{0}.{1}", DateTime.Now.ToShortDateString().Replace(".", "_"), fileFormat);
            result.FileStream.Position = 0;
            
            return result;
        }
        /**/

        public ActionResult ImportSampleDocument() {
            ViewData["SampleDocumentPath"] = SampleDocumentPath;
            HtmlEditorModel model = null;
            HtmlEditorExtension.Import(
                "~/Content/HtmlEditor/SampleImportDocument.rtf",
                HtmlEditorDemosHelper.ImportContentDirectory,
                (html, cssFiles) => model = new HtmlEditorModel(html, cssFiles)
            );
            return View("ImportExport", model);
        }
    }

    public class HtmlEditorModel {
        public HtmlEditorModel(string html) : this(html, null) { }
        public HtmlEditorModel(string html, IEnumerable<string> cssFiles) {
            Html = html;
            CssFiles = cssFiles;
        }

        public string Html { get; set; }
        public IEnumerable<string> CssFiles { get; set; }
    }

    public class HtmlEditorDemosHelper {
        public const string ImportContentDirectory = "~/Content/HtmlEditor/Imported";

        public static HtmlEditorSettings SetHtmlEditorExportSettings(HtmlEditorSettings settings) {
            settings.Name = "heImportExport";
            settings.CallbackRouteValues = new { Controller = "HtmlEditor", Action = "ImportExportPartial" };
            settings.ExportRouteValues = new { Controller = "HtmlEditor", Action = "ExportTo" };

            var toolbar = settings.Toolbars.Add();
            toolbar.Items.Add(new ToolbarUndoButton());
            toolbar.Items.Add(new ToolbarRedoButton());
            toolbar.Items.Add(new ToolbarBoldButton(true));
            toolbar.Items.Add(new ToolbarItalicButton());
            toolbar.Items.Add(new ToolbarUnderlineButton());
            toolbar.Items.Add(new ToolbarStrikethroughButton());
            toolbar.Items.Add(new ToolbarInsertImageDialogButton(true));
            ToolbarExportDropDownButton saveButton = new ToolbarExportDropDownButton(true);
            saveButton.CreateDefaultItems();
            toolbar.Items.Add(saveButton);

            settings.PreRender = (sender, e) => {
                ASPxHtmlEditor editor = (ASPxHtmlEditor)sender;
                editor.Width = Unit.Percentage(100);
            };

            return settings;
        }
    }
}

File: VB/Controllers/HtmlEditorController.vb
Imports System
Imports System.Collections.Generic
Imports System.Web.UI.WebControls
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc
Imports DevExpress.Web.ASPxHtmlEditor

Public Class HtmlEditorController
    Inherits System.Web.Mvc.Controller

    Private Const SampleDocumentPath As String = "~/Content/HtmlEditor/SampleImportDocument.rtf"

    Public Function ImportExport() As ActionResult
        ViewData("SampleDocumentPath") = SampleDocumentPath
        Dim htmlContentPath As String = Server.MapPath("~/Content/HtmlEditor/ImportExport.htm")
        Dim html As String = System.IO.File.ReadAllText(htmlContentPath)
        'ViewData("HTML") = html
        Return View("ImportExport", New HtmlEditorModel(html))
    End Function

    Public Function ImportExportPartial() As ActionResult
        Return PartialView("ImportExportPartial")
    End Function

    Public Function ExportTo(ByVal format As HtmlEditorExportFormat) As ActionResult
        Return HtmlEditorExtension.Export(HtmlEditorDemosHelper.SetHtmlEditorExportSettings(New HtmlEditorSettings()), format)
    End Function

    '
    <HttpPost()> _
    Public Function ExportContentTo() As ActionResult
        Dim format As HtmlEditorExportFormat = EditorExtension.GetValue(Of HtmlEditorExportFormat)("cmbFormat")

        Dim stream As New System.IO.MemoryStream()
        HtmlEditorExtension.Export(HtmlEditorDemosHelper.SetHtmlEditorExportSettings(New HtmlEditorSettings()), stream, format)

        Dim fileFormat As String = format.ToString().ToLower().TrimStart("."c)

        Dim result As New FileStreamResult(stream, "application/" & fileFormat)
        result.FileDownloadName = String.Format("{0}.{1}", DateTime.Now.ToShortDateString().Replace(".", "_"), fileFormat)
        result.FileStream.Position = 0

        Return result
    End Function
    '

    Public Function ImportSampleDocument() As ActionResult
        ViewData("SampleDocumentPath") = SampleDocumentPath
        Dim model As HtmlEditorModel = Nothing

        HtmlEditorExtension.Import(
            "~/Content/HtmlEditor/SampleImportDocument.rtf",
            HtmlEditorDemosHelper.ImportContentDirectory,
            Sub(html, cssFiles)
                model = New HtmlEditorModel(html, cssFiles)
                'ViewData("HTML") = html
            End Sub
        )

        Return View("ImportExport", model)
    End Function
End Class

Public Class HtmlEditorModel
    Public Sub New(ByVal html As String)
        Me.New(html, Nothing)
    End Sub
    Public Sub New(ByVal html As String, ByVal cssFiles As IEnumerable(Of String))
        Me.html = html
        Me.cssFiles = cssFiles
    End Sub

    Private privateHtml As String
    Public Property Html() As String
        Get
            Return privateHtml
        End Get
        Set(ByVal value As String)
            privateHtml = value
        End Set
    End Property
    Private privateCssFiles As IEnumerable(Of String)
    Public Property CssFiles() As IEnumerable(Of String)
        Get
            Return privateCssFiles
        End Get
        Set(ByVal value As IEnumerable(Of String))
            privateCssFiles = value
        End Set
    End Property
End Class

Public Class HtmlEditorDemosHelper
    Public Const ImportContentDirectory As String = "~/Content/HtmlEditor/Imported"

    Public Shared Function SetHtmlEditorExportSettings(ByVal settings As HtmlEditorSettings) As HtmlEditorSettings
        settings.Name = "heImportExport"
        settings.CallbackRouteValues = New With {Key .Controller = "HtmlEditor", Key .Action = "ImportExportPartial"}
        settings.ExportRouteValues = New With {Key .Controller = "HtmlEditor", Key .Action = "ExportTo"}

        Dim toolbar = settings.Toolbars.Add()
        toolbar.Items.Add(New ToolbarUndoButton())
        toolbar.Items.Add(New ToolbarRedoButton())
        toolbar.Items.Add(New ToolbarBoldButton(True))
        toolbar.Items.Add(New ToolbarItalicButton())
        toolbar.Items.Add(New ToolbarUnderlineButton())
        toolbar.Items.Add(New ToolbarStrikethroughButton())
        toolbar.Items.Add(New ToolbarInsertImageDialogButton(True))
        Dim saveButton As New ToolbarExportDropDownButton(True)
        saveButton.CreateDefaultItems()
        toolbar.Items.Add(saveButton)

        settings.PreRender = _
            Sub(sender, e)
                Dim editor As ASPxHtmlEditor = CType(sender, ASPxHtmlEditor)
                editor.Width = Unit.Percentage(100)
            End Sub

        Return settings

    End Function

End Class